examples/00_vanilla/vanilla_example.py

```diff
@@ -0,0 +1,176 @@
+from PIL import Image
+
+DEFAULT_ROTATION_ANGLE = 45
+
+
+def resize_image_to_fill(image, target_width, target_height):
+    img_width, img_height = image.size
+    img_ratio = img_width / img_height
+    target_ratio = target_width / target_height
+
+    if img_ratio > target_ratio:
+        new_width, new_height = target_width, int(target_width / img_ratio)
+    else:
+        new_width, new_height = int(target_height * img_ratio), target_height
+
+    return image.resize((new_width, new_height), Image.LANCZOS)
+
+
+def rotate_image(image, angle):
+    return image.rotate(angle)
+
+
+class ViewAdapter:
+    """
+    Adapter for Generic Remote Controlled Area.
+    """
+
+    def __init__(
+        self,
+        image: Image,
+        name: str,
+        rotation_angle: int,
+        quality: int = 50,
+        **kwargs,
+    ):
+        self.image = image
+        self.area_name = name
+        self.streamer = None
+        self.rotation_angle = rotation_angle
+        self._quality = quality
+        self._press_set = set()
+
+    def _get_metadata(self):
+        return dict(
+            type="image/jpeg",  # supported mime/type
+            codec="",  # video codec, not relevant here
+            w=self.image.size[1],
+            h=self.image.size[0],
+            st=int(time.time_ns() / 1000000),
+            key=("key"),
+            quality=self._quality,
+        )
+
+    def _render(self):
+        fake_file = BytesIO()
+        self.image.save(fake_file, TO_IMAGE_FORMAT["png"], quality=self._quality)
+        content = fake_file.getvalue()
+        self._push(content, self._get_metadata())
+
+    def _push(self, content: bytes, meta: dict):
+        if not self.streamer:
+            return
+        if content is None:
+            return
+        self.streamer.push_content(self.area_name, meta, content)
+
+    def set_streamer(self, stream_manager):
+        self.streamer = stream_manager
+
+    def update_size(self, origin, size):
+        width = max(1, int(size.get("w", 300)))
+        height = max(1, int(size.get("h", 300)))
+        self.image = resize_image_to_fill(self.image, width, height)
+        self._render()
+
+    def on_interaction(self, origin, event):
+        event_type = event["type"]
+        if event_type == "keydown":
+            self.image = rotate_image(self.image, self.rotation_angle)
+        elif event_type == "wheel":
+            delta_y = event.get("deltaY", 0)  # Get scroll direction
+            if delta_y is not None:
+                angle = -self.rotation_angle if delta_y < 0 else self.rotation_angle
+                self.image = rotate_image(self.image, angle)
+            else:
+                return
+        else:
+            return
+        self._render()
+
+
+@TrameApp()
+class VanillaApp:
+    def __init__(self, server=None):
+        self.server = get_server(server, client_type="vue3")
+        image = Image.open("examples/00_vanilla/trame_logo.png")
```

Comment:
You may want to use pathlib.Path to work regardless of the current working directory.